modules/porous_flow/src/materials/PorousFlowRelativePermeabilityUnity.C

```diff
@@ -0,0 +1,40 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "PorousFlowRelativePermeabilityUnity.h"
+#include "Conversion.h"
+
+template<>
+InputParameters validParams<PorousFlowRelativePermeabilityUnity>()
+{
+  InputParameters params = validParams<Material>();
+  params.addRequiredParam<unsigned int>("phase", "The phase number for which to calculate the relative permeability for");
+  params.addRequiredParam<UserObjectName>("PorousFlowDictator_UO", "The UserObject that holds the list of PorousFlow variable names");
+  params.addClassDescription("Base class for PorousFlow relative permeability materials.  This class sets the relative permeability = 1");
+  return params;
+}
+
+PorousFlowRelativePermeabilityUnity::PorousFlowRelativePermeabilityUnity(const InputParameters & parameters) :
+    DerivativeMaterialInterface<Material>(parameters),
+
+    _phase_num(getParam<unsigned int>("phase")),
+    _dictator_UO(getUserObject<PorousFlowDictator>("PorousFlowDictator_UO")),
+    _saturation_variable_name(_dictator_UO.saturationVariableNameDummy()),
+    _saturation_nodal(getMaterialProperty<std::vector<Real> >("PorousFlow_saturation_nodal")),
+    _relative_permeability(declareProperty<Real>("PorousFlow_relative_permeability" + Moose::stringify(_phase_num))),
+    _drelative_permeability_ds(declarePropertyDerivative<Real>("PorousFlow_relative_permeability" + Moose::stringify(_phase_num), _saturation_variable_name))
+{
+  if (_phase_num >= _dictator_UO.numPhases())
```

Comment:
this check seems to be made frequently. You might even add a method in the Dictator object (`checkPhaseNum(_phase_num)`) or have a material base class that owns the `_phase_num` and `PorousFlowDictator_UO` parameters and does the check. 

On that note I tried to simplify parameter names in phase field and _not_ have mixed case and long typenames in them. So this parameter would just be called `dictator ` in phase field. The fact that it is a user object name is already captured in the parameter type, and the fact that a `PorousFlowDictator` is requested should be in the doc string. But I guess that is a matter of taste.